File: VB/ButtonEditWithPadding/PaddingButtonEditViewInfo.vb
Imports Microsoft.VisualBasic
Imports System.Drawing
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraEditors.Drawing
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors.ViewInfo

Namespace ButtonEditWithPadding
	Friend Class PaddingButtonEditViewInfo
		Inherits ButtonEditViewInfo
		Public Sub New(ByVal item As RepositoryItem)
			MyBase.New(item)
		End Sub

		Public Function AdjustBoundsWithMargins(ByVal boundsToAdjust As Rectangle) As Rectangle
			boundsToAdjust.X += (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Left
			boundsToAdjust.Y += (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Top
			boundsToAdjust.Width -= (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Right * 2
			boundsToAdjust.Height -= (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Bottom * 2

			Return boundsToAdjust
		End Function
		Public Function RevertMarginedBounds(ByVal boundsToRevert As Rectangle) As Rectangle
			boundsToRevert.X -= (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Left
			boundsToRevert.Y -= (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Top
			boundsToRevert.Width += (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Right * 2
			boundsToRevert.Height += (CType(Item, RepositoryItemPaddingButtonEdit)).Padding.Bottom * 2

			Return boundsToRevert
		End Function

		Protected Overrides Sub OnBeforeButtonStateChanged(ByVal info As EditorButtonObjectInfoArgs, ByVal newState As ObjectState, ByVal buttonId As Integer)
			If TypeOf Item Is RepositoryItemPaddingButtonEdit Then
				info.Bounds = AdjustBoundsWithMargins(info.Bounds)

				MyBase.OnBeforeButtonStateChanged(info, newState, buttonId)
				info.Bounds = RevertMarginedBounds(info.Bounds)
			Else
				MyBase.OnBeforeButtonStateChanged(info, newState, buttonId)
			End If
		End Sub
	End Class
End Namespace

File: CS/ButtonEditWithPadding/PaddingButtonEditViewInfo.cs
using System.Drawing;
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors.Drawing;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors.ViewInfo;

namespace ButtonEditWithPadding
{
    class PaddingButtonEditViewInfo : ButtonEditViewInfo
    {
        public PaddingButtonEditViewInfo(RepositoryItem item)
            : base(item)
        {
        }

        public Rectangle AdjustBoundsWithMargins(Rectangle boundsToAdjust)
        {
            boundsToAdjust.X += ((RepositoryItemPaddingButtonEdit)Item).Padding.Left;
            boundsToAdjust.Y += ((RepositoryItemPaddingButtonEdit)Item).Padding.Top;
            boundsToAdjust.Width -= ((RepositoryItemPaddingButtonEdit)Item).Padding.Right * 2;
            boundsToAdjust.Height -= ((RepositoryItemPaddingButtonEdit)Item).Padding.Bottom * 2;

            return boundsToAdjust;
        }
        public Rectangle RevertMarginedBounds(Rectangle boundsToRevert)
        {
            boundsToRevert.X -= ((RepositoryItemPaddingButtonEdit)Item).Padding.Left;
            boundsToRevert.Y -= ((RepositoryItemPaddingButtonEdit)Item).Padding.Top;
            boundsToRevert.Width += ((RepositoryItemPaddingButtonEdit)Item).Padding.Right * 2;
            boundsToRevert.Height += ((RepositoryItemPaddingButtonEdit)Item).Padding.Bottom * 2;

            return boundsToRevert;
        }

        protected override void OnBeforeButtonStateChanged(EditorButtonObjectInfoArgs info, ObjectState newState, int buttonId)
        {
            if ( Item is RepositoryItemPaddingButtonEdit )
            {
                info.Bounds = AdjustBoundsWithMargins(info.Bounds);

                base.OnBeforeButtonStateChanged(info, newState, buttonId);
                info.Bounds = RevertMarginedBounds(info.Bounds);
            }
            else
                base.OnBeforeButtonStateChanged(info, newState, buttonId);
        }
    }
}